iceoryx_examples/ara_com/ara/runtime.hpp

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_ARA_COM_RUNTIME_HPP
+#define IOX_EXAMPLES_ARA_COM_RUNTIME_HPP
+
+#include "iceoryx_hoofs/cxx/vector.hpp"
+#include "iceoryx_hoofs/internal/cxx/pair.hpp"
+#include "iceoryx_posh/popo/listener.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "iceoryx_posh/runtime/service_discovery.hpp"
+
+#include "types.hpp"
+
+#include <string>
+
+namespace ara
+{
+class Runtime
+{
+  public:
+    static Runtime& GetInstance(const std::string& name) noexcept
+    {
+        iox::runtime::PoshRuntime::initRuntime(iox::RuntimeName_t(iox::cxx::TruncateToCapacity, name));
+        static Runtime runtime;
+        return runtime;
+    }
+
+    static Runtime& GetInstance() noexcept
+    {
+        static Runtime runtime;
+        return runtime;
+    }
+
+    ara::com::ServiceHandleContainer<com::ProxyHandleType>
+    FindService(ara::com::ServiceIdentifier& serviceIdentifier,
+                ara::com::InstanceIdentifier& instanceIdentifier) noexcept
+    {
+        ara::com::ServiceHandleContainer<com::ProxyHandleType> iceoryxServiceContainer;
+
+        m_discovery.findService(
+            serviceIdentifier,
+            instanceIdentifier,
+            iox::cxx::nullopt,
+            [&](auto& service) {
+                iceoryxServiceContainer.push_back({service.getEventIDString(), service.getInstanceIDString()});
+            },
+            iox::popo::MessagingPattern::PUB_SUB);
+
+        m_discovery.findService(
+            serviceIdentifier,
+            instanceIdentifier,
+            iox::cxx::nullopt,
+            [&](auto& service) {
+                iceoryxServiceContainer.push_back({service.getEventIDString(), service.getInstanceIDString()});
+            },
+            iox::popo::MessagingPattern::REQ_RES);
```

Comment:
That is why a type ANY would have been useful as well. But for verification we may actually need the distinction and not push both kinds in the same container `iceoryxServiceContainer`.

Reply:
`ara::com` does not differentiate between the two communication patterns. So storing everything in one containter is ok. We map them anway with `verifyThatServiceIsComplete`. The service model doesn't fit (anymore).